annotation/tests/test_tasks_services.py

```diff
@@ -1,30 +1,223 @@
 import re
+import uuid
 from copy import deepcopy
-from typing import Dict, List, Optional, Tuple
-from unittest.mock import MagicMock, Mock, call, patch
+from datetime import datetime
+from typing import Dict, List, Optional, Tuple, Union
+from unittest.mock import ANY, MagicMock, Mock, call, patch
 from uuid import uuid4
 
 import pytest
 from fastapi import HTTPException
+from tenant_dependency import TenantData
 
 from annotation.errors import FieldConstraintError
 from annotation.filters import TaskFilter
 from annotation.jobs.services import ValidationSchema
-from annotation.models import File, ManualAnnotationTask
-from annotation.schemas.tasks import ManualAnnotationTaskInSchema
-from annotation.tasks.services import (
-    check_cross_annotating_pages,
-    create_annotation_task,
-    filter_tasks_db,
-    read_annotation_task,
-    read_annotation_tasks,
-    remove_additional_filters,
-    validate_files_info,
-    validate_ids_and_names,
-    validate_task_info,
-    validate_user_actions,
-    validate_users_info,
+from annotation.models import (
+    AgreementMetrics,
+    AnnotatedDoc,
+    File,
+    ManualAnnotationTask,
 )
+from annotation.schemas.tasks import (
+    AgreementScoreServiceResponse,
+    ManualAnnotationTaskInSchema,
+    ResponseScore,
+    TaskStatusEnumSchema,
+)
+from annotation.tasks import services
+
+
+@pytest.fixture
+def mock_task_revisions():
+    yield AnnotatedDoc(
+        pages={"1": ["data1"], "2": ["data2"]},
+        failed_validation_pages=[1, 2],
+        validated=[1, 2],
+    )
+
+
+@pytest.fixture
+def mock_metric():
+    yield AgreementMetrics(
+        task_from=datetime(2024, 1, 1),
+        task_to=datetime(2024, 2, 1),
+        agreement_metric=True,
+    )
+
+
+@pytest.fixture
+def mock_task():
+    yield ManualAnnotationTask(
+        id=1,
+        user_id=2,
+        job_id=10,
+        file_id=3,
+        pages={1, 2, 3},
+        is_validation=False,
+        status=None,
+    )
+
+
+@pytest.fixture
+def mock_stats(
+    mock_task: ManualAnnotationTask, mock_metric: ManualAnnotationTask
+):
+    stat1 = AnnotatedDoc()
+    stat1.task = mock_task
+    stat1.task_id = 1
+    stat1.created = datetime(2024, 1, 1, 12, 0, 0)
+    stat1.updated = datetime(2024, 1, 2, 12, 0, 0)
+
+    stat2 = AnnotatedDoc()
+    stat2.task = mock_task
+    stat2.task_id = 2
+    stat2.created = datetime(2024, 1, 3, 12, 0, 0)
+    stat2.updated = datetime(2024, 1, 4, 12, 0, 0)
+
+    stat3 = AnnotatedDoc()
+    stat3.task = mock_task
+    stat3.task_id = 3
+    stat3.created = datetime(2024, 1, 5, 12, 0, 0)
+    stat3.updated = datetime(2024, 1, 6, 12, 0, 0)
+    stat3.task.status = TaskStatusEnumSchema.finished
```

Comment:
add this data in the constructor

Reply:
It prints an error when I try it. I wrote about it in 910 PR.

Reply:
I saw it, anyway you should create objects using constructor. Check the AnnotatedDoc attributes. AnnotatedDoct doesn't have updated nor created fields

Reply:
Pay attention to the errors you get. You shouldn't bypass errors just because they occur, try to understand why and go through a normal flow. Otherwise we don't test anything